### PR TITLE
opencv: enable Java

### PR DESCRIPTION
**Pull request by valentinegb**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

This pull request changes the `-DBUILD_opencv_java=OFF` CMake option to `-DBUILD_opencv_java=ON`, allowing OpenCV installed from Homebrew to be depended on in Java projects.

### CODE REVIEW

**Comment by valentinegb** [comment]
The [macOS 14-x86_64](https://github.com/Homebrew/homebrew-core/actions/runs/6576262357/job/17865334895) job failed, but all of its steps passed, so I'm not sure what happened there.

**Comment by Bo98** [comment]
I don't think `BUILD_opencv_java` does anything: https://github.com/opencv/opencv/commit/a9163a53d3734babe362049d121b3ef18f46ae85

**Comment by valentinegb** [comment]
Oh, that's super weird, I'm working on a project that depends on the Java version of OpenCV and it didn't work until I set `BUILD_opencv_java` to `ON` in this formula. I'll have to investigate this next time I'm at my computer.

File: Formula/o/opencv.rb [hunk 1]
```diff
@@ -172,5 +177,18 @@ def install
...
+      import org.opencv.core.Core;
+
+      public class Test {
+        public static void main(String[] args) {
+          System.out.print(Core.VERSION);
+        }
+      }
+    EOS
+
+    output = shell_output("#{openjdk}/java -cp #{share}/java/opencv#{version.major}/opencv-#{version.to_s.delete "."}.jar Test.java")
```
Comment (SMillerDev) [inline Formula/o/opencv.rb:191]:
```suggestion
    output = shell_output("#{Formula["openjdk"].opt_bin}/java -cp #{share}/java/opencv#{version.major}/opencv-#{version.to_s.delete "."}.jar Test.java")
```

Comment (valentinegb) [reply Formula/o/opencv.rb]: Didn't know that was a thing, thanks!

**Comment by valentinegb** [comment]
Might need a little bit of help with the test, it passes on my Mac but not in the check.

**Comment by cho-m** [comment]
From logs, I think the build is having trouble finding JNI which may be disabling feature:
```
-- Found Java: /opt/homebrew/opt/openjdk/bin/java (found version "21.0.1") 
-- Could NOT find JNI (missing: AWT JVM) 

...

--   Java:                          
--     ant:                         NO
--     Java:                        YES (ver 21.0.1)
--     JNI:                         /opt/homebrew/opt/openjdk/include /opt/homebrew/opt/openjdk/include
--     Java wrappers:               NO
--     Java tests:                  NO
```

**Pull request closed without merging**